src/modal.jsx

```diff
@@ -1,65 +1,50 @@
-import { Component } from 'react';
 import PropTypes from 'prop-types';
-import HigherOrderPopupComponent from 'rrp/higher-order-popup-component';
-import { TYPE_MODAL } from 'rrp/popup-collection';
-
-const PROP_TYPES = {
+import { connect } from 'react-redux';
+import ModalComponent from './modal-component';
+import { popupSelector } from './popup';
+import { getPortalElement } from './portal';
+import TransitionWrapper from './transition-wrapper';
+
+export const Modal = props => (
+    <TransitionWrapper
+        getPortalElement={props.getPortalElement}
+        isOpen={!!props[props.id]}
+        isPortalReady={props.isPortalReady}
+        render={() => (
+            <ModalComponent
+                className={props.className}
+                layoverClassName={props.layoverClassName}
+                render={props.render}
+                style={props.style}
+            />
+        )}
+        transitionEnterTimeout={props.transitionEnterTimeout}
+        transitionExitTimeout={props.transitionExitTimeout}
+        transitionName={props.transitionName}
+    />
+);
+
+Modal.propTypes = {
+    className: PropTypes.string,
+    getPortalElement: PropTypes.func.isRequired,
     id: PropTypes.string.isRequired,
+    isPortalReady: PropTypes.bool.isRequired,
     layoverClassName: PropTypes.string,
-    popupClassName: PropTypes.string,
-    style: PropTypes.object
+    render: PropTypes.func.isRequired,
+    style: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+    transitionEnterTimeout: PropTypes.number.isRequired,
+    transitionExitTimeout: PropTypes.number.isRequired,
+    transitionName: PropTypes.string.isRequired,
```

Comment:
This forbids passing an object with transition names, which is an important use case with CSS modules.

See https://reactcommunity.org/react-transition-group/#CSSTransition-prop-classNames

Reply:
@Kovensky Thanks, I didn't realize `transitionName` could be configured as an object.
Would this be the right type check?

```javascript
transitionName: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
        appear: PropTypes.string,
        appearActive: PropTypes.string,
        enter: PropTypes.string,
        enterActive: PropTypes.string,
        exit: PropTypes.string,
        exitActive: PropTypes.string
    })
]),

Reply:
Fixed #24